### PR TITLE
Fix nonworking oc command

### PR DESCRIPTION
**Pull request by petrkotas**
**What this PR does / why we need it**:

Fixes non-working oc command in integration tests.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

---
This change is [Reviewable](https://reviewable.io/reviews/kubevirt/kubevirt/2500)

### CODE REVIEW

File: hack/functests.sh [hunk 1]
```diff
@@ -39,7 +39,7 @@ previous_release_registry=${PREVIOUS_RELEASE_REGISTRY:-$_default_previous_releas
 
 functest_docker_prefix=${manifest_docker_prefix-${docker_prefix}}
 
-if [[ ${TARGET} == openshift* ]]; then
+if [[ ${KUBEVIRT_PROVIDER} == os-* ]]; then
```
Comment (davidvossel) [inline hack/functests.sh]: needs to account for 'okd-*' as well.

i'm confused how this ever worked.

Comment (petrkotas) [reply hack/functests.sh]: Thanks @davidvossel I forgot about okd. It worked some time ago, when `TARGET` pointed to images used for current build.

It changed over time, and our `oc` tests were skipped. It does not fail the tests, so it got missed for a while.

Comment (rmohr) [reply hack/functests.sh]: That is because @petrkotas invoked the automation script by hand without exporting the TARGET environment variable.

Comment (petrkotas) [reply hack/functests.sh]: @rmohr that implies the `TARGET` should also be set for our CI, which is not. Due to this, CI is skipping the `oc` tests, which also covers the tests for RHEL images on OCP.

Please see for instance this [PR](https://prow.apps.ovirt.org/view/gcs/kubevirt-prow/pr-logs/pull/kubevirt_kubevirt/2395/pull-kubevirt-e2e-os-3.11.0/1151472742176919554), look at skipped tests. Particular example is: "Templates Creating VM from Template with Fedora Template should succeed to create VirtualMachine via oc command". This test is skipped due to missing `oc` command.

This PR is aimed at fixing this, but I agree setting the variable might be a better solution. I am opening an issue #2505 for it.

Once the issue is foxed, this PR is irrelevant.

Comment (slintes) [reply hack/functests.sh]: TARGET is set by Prow, but IIUC you still need to check for os-* and okd-* instead of openshift-*

Comment (davidvossel) [reply hack/functests.sh]: is KUBEVIRT_PROVIDER appropriate here, or should we still be testing against TARGET?

Comment (slintes) [reply hack/functests.sh]: oh, we are in `hack/functest.sh` here. So it should be `KUBEVIRT_PROVIDER`.
`TARGET` should be used in `automation/test.sh` only, the entrypoint for prow jobs. There we set `KUBEVIRT_PROVIDER=$TARGET`, with 1 special case for the windows target.
Thanks for asking ;)

**Comment by petrkotas** [comment]
@slintes thank you for your help and clarification of how the `TARGET` is really used.
I did accidental commit I had to fix, so the test are run again. I would wait for them.

Otherwise, the #2505 is deprecated. I am closing it.

Again thanks for the help.

**Pull request merged**